editor/src/core/model/element-template-utils.ts

```diff
@@ -565,19 +565,46 @@ export function insertJSXElementChild(
                 const simpleValue = jsxSimpleAttributeToValue(clauseValue)
                 return foldEither(
                   () => {
+                    // Not a simple value, so replacing it would be a bad thing.
                     return clauseValue
                   },
                   (value) => {
+                    // Simple value of some kind.
                     if (value == null) {
+                      // Simple value is null, so replace it with the new content.
                       return elementToInsert
                     } else {
+                      // FIXME: We have a simple `JSXAttribute` value here which is not null/undefined.
+                      // Recent conversations have been about unifying attributes and arbitrary JSX elements,
+                      // which could result in this content being a valid child of any element.
                       return clauseValue
                     }
                   },
                   simpleValue,
                 )
               } else {
-                return clauseValue
+                if (isJSXFragment(clauseValue)) {
```

Comment:
Shouldn't we accept other children affecting elements too and not just fragments?

Reply:
That would be reparenting into those elements though, which is possible to do by dragging into those within the conditional.